src/main/java/org/tron/program/SolidityNode.java

```diff
@@ -23,157 +24,221 @@
-          //dbManager.pushBlock(blockCapsule);
-          for (TransactionCapsule trx : blockCapsule.getTransactions()) {
-            TransactionInfoCapsule ret;
-            try {
-              ret = dbManager.getTransactionHistoryStore().get(trx.getTransactionId().getBytes());
-            } catch (BadItemException ex) {
-              logger.warn("", ex);
-              continue;
-            }
-            ret.setBlockNumber(blockCapsule.getNum());
-            ret.setBlockTimeStamp(blockCapsule.getTimeStamp());
-            dbManager.getTransactionHistoryStore().put(trx.getTransactionId().getBytes(), ret);
-          }
-          dbManager.getDynamicPropertiesStore()
-              .saveLatestSolidifiedBlockNum(lastSolidityBlockNum + 1);
-        } catch (AccountResourceInsufficientException e) {
-          throw new BadBlockException("validate AccountResource exception");
-        } catch (ValidateScheduleException e) {
-          throw new BadBlockException("validate schedule exception");
-        } catch (ValidateSignatureException e) {
-          throw new BadBlockException("validate signature exception");
-        } catch (ContractValidateException e) {
-          throw new BadBlockException("ContractValidate exception");
-        } catch (ContractExeException e) {
-          throw new BadBlockException("Contract Execute exception");
-        } catch (TaposException e) {
-          throw new BadBlockException("tapos exception");
-        } catch (DupTransactionException e) {
-          throw new BadBlockException("dup exception");
-        } catch (TooBigTransactionException e) {
-          throw new BadBlockException("too big exception");
-        } catch (TooBigTransactionResultException e) {
-          throw new BadBlockException("too big exception result");
-        } catch (TransactionExpirationException e) {
-          throw new BadBlockException("expiration exception");
-        } catch (BadNumberBlockException e) {
-          throw new BadBlockException("bad number exception");
-        } catch (NonCommonBlockException e) {
-          throw new BadBlockException("non common exception");
-        } catch (ReceiptCheckErrException e) {
-          throw new BadBlockException("OutOfSlotTime Exception");
-        } catch (VMIllegalException e) {
-          throw new BadBlockException(e.getMessage());
-        } catch (UnLinkedBlockException e) {
-          throw new BadBlockException("unLink exception");
+  private long getLastSolidityBlockNum() {
+    while (true) {
+      try {
+        long blockNum = databaseGrpcClient.getDynamicProperties().getLastSolidityBlockNum();
+        logger.info("Get last remote solid blockNum: {}.", remoteLastSolidityBlockNum);
+        return blockNum;
+      } catch (Exception e) {
+        logger.error("Failed to get last solid blockNum: {}.", remoteLastSolidityBlockNum);
+        sleep(100);
+      }
+    }
+  }
+
+  private void pushBlock() {
+    while (true) {

```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Add an end condition to this loop. ([squid:S2189](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2189))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1570&issues=AWYfmDct_XXc3YMrl43N&open=AWYfmDct_XXc3YMrl43N)

[ISSUE_KEY]: AWYfmDct_XXc3YMrl43N